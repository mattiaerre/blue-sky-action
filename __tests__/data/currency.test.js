const currency = require('../../src/data/currency');
const { BLACK, WHITE } = require('../../src/data/maxContrast');

// SOURCE: https://www.coinbase.com/price

test('validate', () => {
  Object.keys(currency).forEach(key => {
    expect(currency[key].color).toEqual(currency[key].color.toUpperCase());
    expect([BLACK, WHITE].includes(currency[key].foregroundColor)).toBe(true);
    expect({ key, name: currency[key].name }).toMatchSnapshot();
  });
});
