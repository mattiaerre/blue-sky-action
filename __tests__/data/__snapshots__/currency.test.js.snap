// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`validate 1`] = `
Object {
  "key": "ATOM",
  "name": "Cosmos",
}
`;

exports[`validate 2`] = `
Object {
  "key": "BAT",
  "name": "Basic Attention Token",
}
`;

exports[`validate 3`] = `
Object {
  "key": "BCH",
  "name": "Bitcoin Cash",
}
`;

exports[`validate 4`] = `
Object {
  "key": "BTC",
  "name": "Bitcoin",
}
`;

exports[`validate 5`] = `
Object {
  "key": "BSV",
  "name": "Bitcoin SV",
}
`;

exports[`validate 6`] = `
Object {
  "key": "CAD",
  "name": "Canadian Dollar",
}
`;

exports[`validate 7`] = `
Object {
  "key": "DAI",
  "name": "Dai",
}
`;

exports[`validate 8`] = `
Object {
  "key": "DASH",
  "name": "Dash",
}
`;

exports[`validate 9`] = `
Object {
  "key": "EOS",
  "name": "EOS",
}
`;

exports[`validate 10`] = `
Object {
  "key": "ETC",
  "name": "Ethereum Classic",
}
`;

exports[`validate 11`] = `
Object {
  "key": "ETH",
  "name": "Ethereum",
}
`;

exports[`validate 12`] = `
Object {
  "key": "EUR",
  "name": "Euro",
}
`;

exports[`validate 13`] = `
Object {
  "key": "GBP",
  "name": "British Pound",
}
`;

exports[`validate 14`] = `
Object {
  "key": "JPY",
  "name": "Japanese Yen",
}
`;

exports[`validate 15`] = `
Object {
  "key": "KNC",
  "name": "Kyber Network",
}
`;

exports[`validate 16`] = `
Object {
  "key": "LINK",
  "name": "Chainlink",
}
`;

exports[`validate 17`] = `
Object {
  "key": "LTC",
  "name": "Litecoin",
}
`;

exports[`validate 18`] = `
Object {
  "key": "OXT",
  "name": "Orchid",
}
`;

exports[`validate 19`] = `
Object {
  "key": "REP",
  "name": "Augur",
}
`;

exports[`validate 20`] = `
Object {
  "key": "SAI",
  "name": "Single Collateral Dai",
}
`;

exports[`validate 21`] = `
Object {
  "key": "USDC",
  "name": "USD Coin",
}
`;

exports[`validate 22`] = `
Object {
  "key": "XLM",
  "name": "Stellar Lumens",
}
`;

exports[`validate 23`] = `
Object {
  "key": "XRP",
  "name": "XRP",
}
`;

exports[`validate 24`] = `
Object {
  "key": "XTZ",
  "name": "Tezos",
}
`;

exports[`validate 25`] = `
Object {
  "key": "ZEC",
  "name": "Zcash",
}
`;

exports[`validate 26`] = `
Object {
  "key": "ZRX",
  "name": "0x",
}
`;
